src/server/worker/pipeline/transform/registry.go

```diff
@@ -441,32 +440,38 @@ func (reg *registry) processJobRunning(pj *pendingJob) error {
 			return err
 		}
 	}
-
-	// TODO: This is a hack to ensure that deletions are generated before any output is uploaded.
-	// This may be resolved by either explicitly generating deletes first (somewhat similar to this hack) or
```

Comment:
So we are now doing the first method, which is "similar to this hack" ?

Reply:
Yeah, I kind of considered it a hack since it was the only way to ensure the correct ordering of deletes when workers were operating directly on the commits. It seems like a reasonable approach now because we have a path forward for caching the computation of the datums to process and in a lot of cases we need to know the number of datums to determine how to divide them into datumsets (when a user does not set a datumset spec). Also, it allows us to immediately add the fileset IDs to the commits since the deletions will already be applied.